lib/detect-libc.js

```diff
@@ -47,7 +47,7 @@ const GLIBC = 'glibc';
  * A Regexp constant to get the GLIBC Version.
  * @type {string}
  */
-const RE_GLIBC_VERSION = /GLIBC\s(\d+\.\d+)/;
+const RE_GLIBC_VERSION = /LIBC.*?(\d+\.\d+)/i;
```

Comment:
It looks like this matcher might now be too loose as it is interpreting the contents of `/usr/bin/ldd` on musl-based Void Linux as if the version is 2.35, which it is not.

> ...startlibc_start_main_stage2argv__hwcaplsm2_fnrevents__progname_fullsrc/env/__reset_tls.c__reset_tls__init_sspsrc/env/__stack_chk_fail.c__stack_chk_fail__stack_chk_guardclearenvsrc/env/clearenv.csrc/env/getenv.c__strchrnulunsetenvsrc/env/putenv.coldenvnewenv__putenvsrc/env/secure_getenv.csecure_getenvenv_allocedenv_alloced_nsrc/env/setenv.coverwritestrlen__env_rm_addsrc/env/unsetenv.csrc/errno/__errno_location.csrc/errno/strerror.c__lctranserriderrmsg__strerror_l_Exitsrc/exit/_Exit.craisesrc/exit/abort.c__abort_lockk_sigaction__block_all_sigs__syscall4abort__assert_fail_IO_FILEfflushsrc/exit/assert.c__stderr_FILEat_quick_exitsrc/exit/at_quick_exit.c__funcs_on_quick_exit__cxa_atexitsrc/exit/atexit.cbuiltin__cxa_finalizenew_fl__funcs_on_exit__fini_array_startsrc/exit/exit.c__fini_array_endsrc/exit/quick_exit.ccreatfilenamesrc/fcntl/creat.c__builtin_va_listfp_offsetf_owner_ex__syscall_cpgp_offsetreg_save_area__syscall_retoverflow_arg_areasrc/fcntl/fcntl.c__va_list_tagsrc/fcntl/open.csrc/fcntl/openat.copenatadvicesrc/fcntl/posix_fadvise.cposix_fadviseposix_fallocatesrc/fcntl/posix_fallocate.csrc/fenv/__flt_rounds.c__flt_roundsfegetroundsrc/fenv/fegetexceptflag.cfegetexceptflagfexcept_tfetestexcept__unused4__unused5__control_wordfenv_t__data_selectorfeholdexceptfeclearexceptfegetenv__mxcsr__opcode__data_offsetsrc/fenv/feholdexcept.c__tags__eip__cs_selector__status_word__unused1__unused2__unused3feraiseexceptfesetexceptflagsrc/fenv/fesetexceptflag.csrc/fenv/fesetround.c__fesetroundfeupdateenvfesetenvsrc/fenv/feupdateenv.csrc/fenv/x86_64/fenv.s.GNU AS 2.35.1src...

Perhaps replace the `.*` with something a bit tighter? The above output is from `grep -a` and will be missing zero bytes so maybe the regex could check for only "printable" characters between "libc" and the version number?

Reply:
Decided to go with `a-z0-9 \-).` since that seemed to be the tightest set of characters that still matched on libc and glibc distros without also matching on Void Linux's musl binary `/usr/bin/ldd`.